### ISSUE TITLE
Google Ads Integration Version Deprecated

### ISSUE THREAD

**Issue by Lee4cj** [mage-ai/mage-ai#5525]
When setting up the google ads integration, it throws an error that version 15 is depreciated.

 request_error: UNSUPPORTED_VERSION\n  }\n  message: \"Version v15 is deprecated. Requests to this version will be blocked.

**Comment by Rykarix** [comment]
I can't use google ads because:
google has depreciated all versions below v18, see here for depreciation details
mage-ai==0.9.74 depends on protobuf>=4.21.12,<4.22.dev0
google-ads==25.1.0 depends on protobuf>=4.25.0,<6.0.0

this means that google-ads is locked to 22.1.0 which only supports v15 (which google will reject as a valid request in the response)

Google Ads api v16,17 and 18 earliest compatable protobuff version is 4.25.0. You'll need to update the dependancies accordingly

@tommydangerous @wangxiaoyou1993

**Comment by csharplus** [comment]
@Rykarix please see if the changes in https://github.com/mage-ai/mage-ai/pull/5601 work for you.

**Comment by Rykarix** [comment]
@csharplus I don't have a contributors setup running anymore.

Ultimately, as long as mage isn't constricted to `protobuf>=4.21.12,<4.22.dev0` then it'll fix it. After reviewing your code, I can't see this working unless you modify the following requirements files:

mage_integrations/requirements.txt > line 32 > `protobuf>=4.0.0` needs to be changed to `protobuf>=4.25.0`
requirements.txt > line 32 > `protobuf~=4.21.12` needs to be changed to `protobuf>=4.25.0`

Then, you need to obviously update any library that requires `protobuf<=4.25.0`

**Comment by Rykarix** [comment]
@wangxiaoyou1993 @tommydangerous 

Also, I think this further highlights the benefits of moving away from `requirements.txt` and towads using pyproject.yoml.

I'm using UV on my local build and it would help tremendously when it comes to dependancy management as it will simply tell you why it wont sync versions.

Consider the following:
```toml
dependencies = [
    "google-ads>=24.0.0",
]

[tool.uv]
dev-dependencies = [
    "ruff>=0.7.0",
    "mage-ai>=0.9.74",
    "pyodbc>=5.2.0",
    "plotly>=5.24.1",
    "panel>=1.5.4",
    "watchfiles>=0.24.0",
    "jupyter-bokeh>=4.0.5",
    "pydantic>=2.7.4",
]
```
```bash
 uv sync
```

Output:
```
  × No solution found when resolving dependencies for split (python_full_version >= '3.10.15' and python_full_version < '3.11'):
  ╰─▶ Because only the following versions of google-ads are available:
          google-ads<=24.0.0
          google-ads==24.1.0
          google-ads==25.0.0
          google-ads==25.1.0
      and google-ads>=24.0.0,<=24.1.0 depends on protobuf>=4.25.0,<5.0.0, we can conclude that google-ads>=24.0.0,<25.0.0 depends on protobuf>=4.25.0,<5.0.0.
      And because google-ads>=25.0.0 depends on protobuf>=4.25.0,<6.0.0 and mage-ai==0.9.74 depends on protobuf>=4.21.12,<4.22.dev0, we can conclude that google-ads>=24.0.0 and mage-ai==0.9.74 are incompatible.
      And because only mage-ai<=0.9.74 is available and mtm-mageai:dev depends on mage-ai>=0.9.74, we can conclude that google-ads>=24.0.0 and mtm-mageai:dev are incompatible.
      And because your project depends on google-ads>=24.0.0 and mtm-mageai:dev, we can conclude that your project's requirements are unsatisfiable.
```

As you can imagine, this would be very helpful for you to easily maintain your dependancies as you'll effectively get a report of exactly what the conflicts are

**Comment by csharplus** [comment]
@Rykarix very helpful information! Some additional updates were made per your suggestions. Thanks.
